website/docs/r/cosmosdb_sql_role_definition.html.markdown

```diff
@@ -0,0 +1,102 @@
+---
+subcategory: "CosmosDB (DocumentDB)"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_cosmosdb_sql_role_definition"
+description: |-
+  Manages a Cosmos DB SQL Role Definition.
+---
+
+# azurerm_cosmosdb_sql_role_definition
+
+Manages a Cosmos DB SQL Role Definition.
+
+## Example Usage
+
+```hcl
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_cosmosdb_account" "example" {
+  name                = "example-cosmosdb"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  offer_type          = "Standard"
+  kind                = "GlobalDocumentDB"
+
+  consistency_policy {
+    consistency_level = "Strong"
+  }
+
+  geo_location {
+    location          = azurerm_resource_group.example.location
+    failover_priority = 0
+  }
+}
+
+resource "azurerm_cosmosdb_sql_role_definition" "example" {
+  name                = "84cf3a8b-4122-4448-bce2-fa423cfe0a15"
+  resource_group_name = azurerm_resource_group.example.name
+  account_name        = azurerm_cosmosdb_account.example.name
+  role_name           = "acctestsqlrole"
+  assignable_scopes   = ["/subscriptions/${data.azurerm_client_config.current.subscription_id}/resourceGroups/${azurerm_resource_group.example.name}/providers/Microsoft.DocumentDB/databaseAccounts/${azurerm_cosmosdb_account.example.name}/dbs/sales"]
+
+  permissions {
+    data_actions = ["Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/items/read"]
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The GUID as the name of the Cosmos DB SQL Role Definition. Changing this forces a new resource to be created.
+
```

Comment:
Hi @neil-yechenwei! Thanks, looking forward to seeing this implemented and released! I looked over the code quickly, will circle back as soon as I've a bit more time on my hands.

For now my main question is about naming in comparison to `azurerm_role_definition`. I looked at [`azurerm_role_definition` as a comparable resource](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/role_definition#argument-reference) and was wondering if it is better to build `azurerm_cosmos_sql_role_definition` in a similar way.

`azurerm_role_definition.name` is in that case comparable to `azurerm_cosmos_sql_role_definition.role_name`, and `azurerm_role_definition.role_definition_id` to `azurerm_cosmos_sql_role_definition.name`.

Does it make sense to use it in a similar way here, or is this something different? From a user perspective this is a bit harder to comprehend and confusing if these resources work differently.

Reply:
Thanks for suggestion. I've updated code.